### PR TITLE
Migrate post_groupDelete hook to share manager

### PR DESCRIPTION
**Pull request by rullzer**
For #22209

The hook now calls the share manager that will call the responsible
shareProvider to do the proper cleanup.
- Unit tests added

Again nothing should change it is just to cleanup old code

CC: @schiesbn @nickvergessen @PVince81 @MorrisJobke @DeepDiver1975

### CODE REVIEW

**Comment by rullzer** [comment]
Apparently I do something mysql does not like... I'll look into it

**Comment by rullzer** [comment]
Ok, mysql is happy now as well.
Review time.

File: lib/private/Share20/DefaultShareProvider.php [hunk 1]
```diff
@@ -867,4 +867,49 @@ public function userDeleted($uid, $shareType) {
...
+
+		$cursor = $qb->execute();
+		$ids = [];
+		while($row = $cursor->fetch()) {
+			$ids[] = (int)$row['id'];
+		}
+		$cursor->closeCursor();
+
+		$offset = 0;
+		$slice = array_slice($ids, $offset, 100);
```
Comment (nickvergessen) [inline lib/private/Share20/DefaultShareProvider.php]: array_chunk?

Comment (rullzer) [inline lib/private/Share20/DefaultShareProvider.php]: yes... lets use that indeed..

File: lib/private/Share20/DefaultShareProvider.php [hunk 2]
```diff
@@ -867,4 +867,47 @@ public function userDeleted($uid, $shareType) {
...
+			->andWhere($qb->expr()->eq('share_with', $qb->createNamedParameter($gid)));
+
+		$cursor = $qb->execute();
+		$ids = [];
+		while($row = $cursor->fetch()) {
+			$ids[] = (int)$row['id'];
+		}
+		$cursor->closeCursor();
+
+		if (count($ids) > 0) {
```
Comment (nickvergessen) [inline lib/private/Share20/DefaultShareProvider.php]: `!empty($ids)`

Comment (rullzer) [inline lib/private/Share20/DefaultShareProvider.php]: fixed

**Comment by MorrisJobke** [comment]
Rebased to retrigger CI

**Comment by MorrisJobke** [comment]
Tested and works 👍

File: lib/private/share/hooks.php [hunk 3]
```diff
@@ -161,17 +161,4 @@ public static function post_removeFromGroup($arguments) {
...
-
-	/**
-	 * Function that is called after a group is removed. Cleans up the shares to that group.
-	 * @param array $arguments
-	 */
-	public static function post_deleteGroup($arguments) {
-		$sql = 'SELECT `id` FROM `*PREFIX*share` WHERE `share_type` = ? AND `share_with` = ?';
-		$result = \OC_DB::executeAudited($sql, array(self::SHARE_TYPE_GROUP, $arguments['gid']));
-		while ($item = $result->fetchRow()) {
-			Helper::delete($item['id']);
```
Comment (nickvergessen) [inline lib/private/share/hooks.php]: Can we delete `Helper::delete()`  as well, or is it still used from elsewhere?

Comment (nickvergessen) [inline lib/private/share/hooks.php]: Okay seems to be still used.

**Comment by nickvergessen** [comment]
:+1:

File: lib/private/share/hooks.php (hunk 3 above)
Comment (rullzer) [inline lib/private/share/hooks.php]: yeah not yet... but :soon:

**Pull request merged**